prime-router/docs/proposals/lineage/README.md

```diff
@@ -0,0 +1,123 @@
+
+The Hub implements a simple workflow engine.   Each step in the engine is a **Task**.  Most (but not all) tasks create one or more Reports. Current **Task** in the system include: 
+
+| Task      |     Description
+| ---       | ---
+| Receive   | Receive a new Report into the Hub
+| Transform | Validate a received report, and transform into desired output formats, according to configured data filter business rules
+| Batch     |  Merge one or more reports together.   Typically done to consolidate data per receiver requirements.
+| Send      | Reliably delivery data to an external receiver, eg, an API, sftp site, or provide for upload (and notify with an email)
+| Wipe      | Remove data, per business rules
+
+An **Organization** can recieve and/or send data to the hub.  A typical Organization is a state or local Public Health Department.   Strac, Redox, and SimpleReport are all also Organizations in the hub.
+
+An **OrganizationService** is an endpoint to which the Hub can send a specific kind of data to a specific place, so that the Organiation can use the data.  For example, an OrganizationService might be an sftp site where a local health department wants hl7 files uploaded every 4 hours.
+
+Likewise, an **OrganizationClient** is an endpoint that sends data to the Hub API.   For example, SimpleReport has an OrganizationClient that can send .csv data to the hub, as often as they like.
+
+
+## Assumptions
+
+- TASKs create zero or more REPORTs which have one or more ITEMs
+- ITEMs are like quarks.They only exist inside of a REPORT.  Its fine and quite common to have a report with one item.
+- We must track lineage per-item, however, people think in terms of REPORTs.  Hence we need to track both.
+- Tracking of sending- and receiving- Organizations is crucial, to know who gets permissions to see the data.
+- Many:many -- A given ACTION could take in many reports, and produce many reports.  Need to allow for both splitting and merging.
+- Many:many -- An ITEM can be the result of multiple parent items, and can result in multiple child ITEMs.
+- A few schema best practices:  Every table must have a single col PK (either a serial or a UUID), and a created_date, Normalize independent data.   Use singular nouns.
+-- Receipt of a dataset implies the right to see the lineage of that dataset.
+-- Sending a dataset implies the right to see the lineage of that dataset. 
+
+## Proposed Design
+
+Please [pull up the ER diagram](./lineage-er-diagram.jpeg) for the detailed proposed schema.
+
+(Note: [Original lucid chart diagram is here](https://lucid.app/lucidchart/7b81cccb-91a0-44b3-b17f-90b9b0c8304e/edit?beaconFlowId=332110D5160D6847&page=0_0#))
+
```

Comment:
I would like to track the `receiving_org_results` and `receiving_org_params` like you are proposing for the sending side. Redox, in particular, sends back an ID that I want to track in our DB.

Reply:
I've added this.